### PR TITLE
Enhancement/71 remove client conditional directives from the code

### PR DESCRIPTION
**Pull request by sdvendramini**
Related issue:

https://github.com/wazuh/wazuh-agent/issues/71

## Description

This PR adds the changes to delete the use of the CLIENT conditional directive and remove any unnecessary manager-related code used along with these directives.

### CODE REVIEW

File: src/common/utils/shared.h [hunk 1]
```diff
@@ -221,11 +221,7 @@ extern const char *__local_name;
 // Only static arrays allowed.
 #define array_size(array) (sizeof(array)/sizeof(array[0]))
 
-#ifdef CLIENT
 #define isAgent 1
```
[the remark this answers is not in the document]
Comment (sdvendramini) [reply src/common/utils/shared.h]: Removed. Is not used anymore.

File: src/common/syscheck_op/src/syscheck_op.c [hunk 2]
```diff
@@ -115,429 +115,6 @@ int remove_empty_folders(const char *path) {
 }
 
 #ifndef WIN32
-#ifndef CLIENT
-int sk_decode_sum(sk_sum_t *sum, char *c_sum, char *w_sum) {
```
[the remark this answers is not in the document]
Comment (sdvendramini) [reply src/common/syscheck_op/src/syscheck_op.c:119]: Done!

File: src/common/syscheck_op/include/syscheck_op.h [hunk 3]
```diff
@@ -291,9 +246,6 @@ void normalize_path(char *path);
  * @return A string with escaped characters
  */
 char *escape_syscheck_field(char *field);
-#ifndef CLIENT
-char *unescape_syscheck_field(char *sum);
```
Comment (TomasTurina) [inline src/common/syscheck_op/include/syscheck_op.h:295]: Remove from .c.

**Review by TomasTurina — COMMENTED** [review]
Remove related UT to functions that don't exist anymore.

File: src/common/syscheck_op/include/syscheck_op.h (hunk 3 above)
Comment (sdvendramini) [reply src/common/syscheck_op/include/syscheck_op.h:295]: Done!

**Review by TomasTurina — COMMENTED** [review]
Remove all code under `TEST_SERVER` flags.

**Review by TomasTurina — APPROVED** [review]

**Pull request merged**